syntaxes/haskell.YAML-tmLanguage

```diff
@@ -221,7 +224,13 @@ patterns:
       - include: '#record_decl'
       - include: '#deriving'
       - include: '#comments'
-      - begin: '^(\s*)(?:(\|)|(=))\s*(\b[\p{Lu}\p{Lt}][\p{Ll}_\p{Lu}\p{Lt}\p{Nd}'']*)'
+      - begin: >-
+          (?x)
+            ^(\s*)
+              (?:(\|)|(=))\s*
+              ( (?:\b[\p{Lu}\p{Lt}][\p{Ll}_\p{Lu}\p{Lt}\p{Nd}']*)
+              | (?:\(:[\p{S}\p{P}&&[^(),;\[\]`{}_"']]*\))
+              )\s*
```

Comment:
This (and change just below) adds prefix constructors.

Reply:
Right! :+1: